src/states.jl

```diff
@@ -491,6 +493,87 @@ function PhaseEquil_ρpq(
     return PhaseEquil{FT, typeof(param_set)}(param_set, ρ, e_int, q_tot, T)
 end
 
+
+#####
+##### Anelastic-equilibrium states
+#####
+"""
+    AnelasticPhaseEquil{FT} <: AbstractPhaseEquil{FT}
+
+A thermodynamic state assuming thermodynamic equilibrium (therefore, saturation adjustment
+may be needed).
+
+# Constructors
+
+    PhaseEquil(param_set, ρ, p, e_int, q_tot, T)
+
+# Fields
+
+$(DocStringExtensions.FIELDS)
+"""
+struct AnelasticPhaseEquil{FT, PS} <: AbstractPhaseEquil{FT}
+    "parameter set, used to dispatch planet parameter function calls"
+    param_set::PS
+    "full density of air (potentially moist)"
+    ρ::FT
+    "reference air pressure"
+    p::FT
+    "internal energy"
+    e_int::FT
```

Comment:
Currently we do not need it but if we change the prognostic variable in the EDMF to energy, this will be needed. therefore I left it in.
If you still think it should be removed from the thermodynamic state we can take it.

Reply:
Again, I'd like to get rid of references to anelastic in thermodynamics (because it really doesn't belong there). If we use internal or total energy, and if it turns out we need a new constructor for thermodynamics, we can add it then. But here, all I think we need is a constructor for thermodynamics given pressure, and that does not require a new `PhaseEquil`, just modifications to functions that should take pressure as input (and not compute density along the way).